### DISCUSSION TITLE
Detecting one extra empty column in the table

### DISCUSSION THREAD

**Discussion by kathimohan** [jsvine/pdfplumber#789] [Ask for help with specific PDFs]
pdfplumber detecting extra column in a table by taking start of page to first vertical line as one column but needs to take table from the first vertical line, how to make it possible.

**Comment by jsvine** [comment]
Hi @kathimohan. Without the code or PDF, this is a difficult question to answer. Please provide those, as well as a more detailed explanation of the problem you're trying to solve. Thanks!

**Reply by kathimohan** [reply]
[emp1.pdf](https://github.com/jsvine/pdfplumber/files/10453687/emp1.pdf)
For this file, one empty column before the column having data, but pdfplumber detects two columns for these kind of pages, instead of detecting table from first vertical column, it is detecting from the page boundaries. I am using this line - tb1 = page.extract_tables(), for extracting tables, Is there any way to extract the table from first vertcal column to last vertical column and not considering the page boundaries.

**Reply by jsvine** [reply]
Thanks for the PDF. If I'm understanding the description correctly (though I'm not sure I am), I'd suggest using the `"vertical_strategy": "explicit", "explicit_vertical_lines": [...]` table settings, passing in only specific vertical lines you want to use for table detection.

**Reply by kathimohan** [reply]
.lines property not providing the line dimensions, can u pls suggest how can i get the vertical lines dimensions of a page.

**Reply by jsvine** [reply]
Did you also examine `.rects`? Sometimes PDFs draw "lines" as narrow rectangles.

**Reply by kathimohan** [reply]
but how to find whether rect is horizontal line or vertical line

**Reply by jsvine** [reply]
That should be possible by comparing the `"height"` property to the `"width"` property.

**Reply by kathimohan** [reply]
I didn't get, can u pls give me some idea. This is one vertical lines rect  dimensions - {'x0': 36.0, 'y0': 36.04, 'x1': 576.0, 'y1': 756.0, 'width': 540.0, 'height': 719.96, 'pts': [(36.0, 756.0), (576.0, 756.0), (576.0, 36.04), (36.0, 36.04)], 'linewidth': 0.0375, 'stroke': False, 'fill': True, 'evenodd': False, 'stroking_color': None, 'non_stroking_color': (1, 1, 1), 'object_type': 'rect', 'page_number': 17, 'top': 36.0, 'bottom': 755.96, 'doctop': 12708.0}

**Reply by jsvine** [reply]
Unfortunately, I don't have time to walk through the specifics of this PDF, although it's possible that someone else in the community does. In the meantime, you may find `pdfplumber`'s visual debugging tools useful: https://github.com/jsvine/pdfplumber#visual-debugging

In particular:

```python
page = pdf.pages[0]
im = page.to_image()
im.draw_rects(page.rects)
```
